### ISSUE TITLE
Recognize ConditionalAttribute in BannedApiAnalyzers.

### ISSUE THREAD

**Issue by manfred-brands** [dotnet/roslyn-analyzers#7581]
### Analyzer

**RS0030**: [BannedApiAnalyzers](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Microsoft.CodeAnalysis.BannedApiAnalyzers.md)

### Describe the improvement

We don't want libraries to use the type `Console`, except in Debug or Verbose builds whilst debugging.
So we added a conditional reference to the `Microsoft.CodeAnalysis.BannedApiAnalyzers` with an appropriate `BannedSymbols.txt`
```xml
  <ItemGroup Condition="'$(Configuration)'=='Release'">
    <PackageReference Include="Microsoft.CodeAnalysis.BannedApiAnalyzers" Version="3.3.4" />
    <AdditionalFiles Include="BannedSymbols.txt" />
  </ItemGroup>
```

However we have conditional executed code like:
```csharp
[Conditional("VERBOSE")]
private static void Report(xxxx)
{
    System.Console.WriteLine(xxx);
}
```

I would like to be able to configure the analyzer to exclude reporting usages in methods annotated with the `Conditional` attribute using a specific constant.

### Describe suggestions on how to achieve the rule

When usage of a banned api is detected, the analyzer should check for a `ConditionalAttribute` on the containing method/class and only report if the `ConditionString` doesn't match allowed conditions. Or better if the analyzer knows the predefined symbols doesn't match those.

### Additional context

![Image](https://github.com/user-attachments/assets/16c4e8f8-78f5-44d7-8844-bf07bee6d579)
